operator/flags.go

```diff
@@ -42,6 +42,12 @@ func init() {
 	flags.Float64(option.IPAMAPIQPSLimit, defaults.IPAMAPIQPSLimit, "Queries per second limit when accessing external IPAM APIs")
 	option.BindEnv(option.IPAMAPIQPSLimit)
 
+	flags.String(option.AzureSubscriptionID, "", "Subscription ID to access Azure API")
+	option.BindEnvWithLegacyEnvFallback(option.AzureSubscriptionID, "AZURE_SUBSCRIPTION_ID")
+
+	flags.String(option.AzureResourceGroup, "", "Resource group to use for Azure IPAM")
+	option.BindEnvWithLegacyEnvFallback(option.AzureResourceGroup, "AZURE_RESOURCE_GROUP")
```

Comment:
Why are we binding `AzureSubscriptionID` with the legacy env `AZURE_SUBSCRIPTION_ID`? It's the first time that we are introducing azure so how can this flag be legacy?

Reply:
This is the standard env name as per https://github.com/banzaicloud/azure-aks-client/blob/master/README.md. Both the regular CIlium syntax and the azure-aks-client name are supported with this.

Reply:
Is the confusing part that the function is called legacy? We are basically treating the standardized env name as the legacy part to support both env names.